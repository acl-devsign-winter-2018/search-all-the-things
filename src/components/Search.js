import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './search.css';

export default class Search extends Component {
 
  static propTypes = {
    onSearch: PropTypes.func.isRequired,

  };

  state = {
    search: ''
  };

  handleSubmit = (event) => {
    event.preventDefault();
    const { search } = this.state;
    if(!search.trim()) return;
    this.props.onSearch(search);
  };

  handleChange = ({ target }) => {
    this.setState({ search: target.value });
  };

  render() {
    const { search } = this.state;

    return (
      <form className={styles.search} onSubmit={this.handleSubmit}>
        <label>
          Search For Movies:
          <input value={search} onChange={this.handleChange}/>
        </label>
        <button >Search</button>
      </form>
    );
  }
}
