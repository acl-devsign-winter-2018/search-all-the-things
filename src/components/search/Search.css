:local(.search) {
  text-align: center;
  border: 1px solid black;
  padding: 20px;
  background: grey;
}

input {
  margin: 2%;
  border: 1px solid black;
  font-size: 1.25em;
}

button {
  box-shadow: -5px 5px 5px grey;
  margin-left: 2%;
  font-size: 1em;
  border: 1px solid black;
}

button:active {
  box-shadow: -2px 2px 2px;
}

