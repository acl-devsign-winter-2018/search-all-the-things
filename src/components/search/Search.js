import React, { Component } from 'react'; 
import PropTypes from 'prop-types';
import styles from './Search.css';

export default class Search extends Component {

  static propTypes = {
    onSearch: PropTypes.func
  };

  state = {
    subject: ''
  };

  handleSubmit = (event) => {
    event.preventDefault();
    const { subject } = this.state;
    if(!subject.trim()) return;
    this.props.onSearch({ ...this.state });
  };

  handleChange = ({ target }) => {
    this.setState({ subject: target.value });
  };

  render() {

    const { subject } = this.state;
    
    return (
      <form className={styles.search} onSubmit={this.handleSubmit}>
        <label>
          Search for books by subject:
          <input name="search" value={subject} onChange={this.handleChange}/>
        </label>
        <button>Find</button>

      </form>
    );
  }
}