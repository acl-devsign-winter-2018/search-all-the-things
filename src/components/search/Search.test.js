import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Search from './Search';

describe('Search', () => {

  it('Renders Properly', () => {
    const wrapper = shallow(<Search/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

});
