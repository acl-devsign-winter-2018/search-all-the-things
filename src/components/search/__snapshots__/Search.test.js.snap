// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Renders Properly 1`] = `
<form
  onSubmit={[Function]}
>
  <label>
    Search for books by subject:
    <input
      name="search"
      onChange={[Function]}
      value=""
    />
  </label>
  <button>
    Find
  </button>
</form>
`;
